### DISCUSSION TITLE
当开启惰性加载图片时会引起个别网站图片消失的解决方案

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#2347] [重要通知]
# 最新通知

![image](https://github.com/Kenshin/simpread/assets/81074/3fd24bc9-4c53-43a8-99b4-1bff7d39d87e)

如果你因为访问维基百科导致效果不佳（包括：正文消失）问题时，可以使用 [维基百科辅助增强](https://github.com/Kenshin/simpread/discussions/5832) 彻底解决此问题。

除此之外，此插件还包括如下优化和功能：

- 仅保留正文，使其更转换为 Markdown 结构，去掉了一些其它元素

- 删除表格（在个别维基百科的页面中，表格是非常复杂的结构，所以生成的 Markdown 效果非常差，因此可以删除这些表格）

- 让一些在侧边的图片居中显示

- 可正常显示公式

- 删除页脚

- 清理文中链接（更利于导入到 Obsidian 不会生成多余的双向链接）

- 删除正文的全部链接

### 安装与升级

- 没有【使用同步助手 · 自动同步】的话，可通过 [此链接](https://simpread.ksria.cn/plugins/details/9YsSaZ1uAq) 安装。

- 已经【使用同步助手 · 自动同步】的话，请使用 [插件管理器](https://www.yuque.com/kenshin/simpread/fcw60u#ymS1C) 安装。

详细说明请看这里 https://github.com/Kenshin/simpread/discussions/5832

***

## 问题现象

当访问 **维基百科出现正文无法显示**，则请看下面的解决方案。

## 问题分析

因简悦 2.2 发布了惰性图片加载方案，而维基百科在页面中隐藏了一个无意义的占位图片，导致此问题的发生

![image](https://user-images.githubusercontent.com/81074/121171725-6f669280-c889-11eb-9566-acee07869794.png)

惰性加载方案会提前加载图片，因此会将上段内容替换原本惰性加载的占位，但因为一个 Bug 造成了这问题。

2.2.0.520 已经修复了 `wikipedia.org` 但维基旗下的其它服务（下面会介绍）则仍存在此问题。（同样解决方案也在下方）

### 2.2.0.520 用户（方案一）

请打开选项页 → 高级设定 → 词法分析引擎 → 关闭此选项

![image](https://user-images.githubusercontent.com/81074/120952856-6f2da080-c77e-11eb-8e1a-0f6d9da8e40f.png)

### 或将下面的域名加入黑名单（方案二）

发现 **Wiki 旗下的页面均有可能受到影响**，如不想禁用 **惰性加载解析** 方案的话，请将下面的内容加入黑名单：

```
wiktionary.org
wikimedia.org
wikiquote.org
wikinews.org
wikivoyage.org
mediawiki.org
wikisource.org
wikibooks.org
wikiversity.org
wikidata.org
```

#### 方案一

> 将  `wikipedia.org` 加入 **黑名单**，即可。

> 使用此方式后，无法在 维基百科 下使用简悦。

![image](https://user-images.githubusercontent.com/81074/118933827-40d95400-b97c-11eb-9ca0-23da0ce3b35d.png)

#### 方案二

> 将  `wikipedia.org` 加入 **延迟加载列表**，对有需要的页面加入稍后读（在稍后读的阅读模式下，可以查看

![image](https://user-images.githubusercontent.com/81074/118934526-07edaf00-b97d-11eb-8ba0-5507adc548fd.png)

![image](https://user-images.githubusercontent.com/81074/118934555-0e7c2680-b97d-11eb-91db-0a69e2b89764.png)

**Comment by Kenshin** [comment]
关联问题 https://github.com/Kenshin/simpread/issues/3450

**Comment by eromoe** [comment]
添加一个站点， 子域名上的文章都受影响
mirror.xyz

**Reply by Kenshin** [reply]
谢谢反馈~

PS：使用方案一不需要加名单的。

**Comment by felix2049lou** [comment]
> 没有【使用同步助手 · 自动同步】的话，可通过 [此链接](https://simpread.ksria.cn/plugins/details/N3EznnUXeV) 安装。

連接不是https://simpread.ksria.cn/plugins/details/9YsSaZ1uAq

**Reply by Kenshin** [reply]
🙏 

已修复 😂
